rfcs/0000-wire-policies-with-backend-services.md

```diff
@@ -0,0 +1,181 @@
+The design assumed one policy controller instance per limitador instance. The policy controller got limitador's location at boot time via an environment variable.
+After the policies merge into the operator's pod, the policies controllers will be a singleton instance (one pod, one container) at the cluster level.
+Regardless of the kuadrant's support for multiple or single limitador/authorino instances, the policies controllers will be running in a single pod in the entire cluster.
+Even if kuadrant only supports a single limitador/authorino instance,
+the policies controllers still need to know the location of the limitador/authorino instances.
+
+Therefore, a new design is needed that wires the user created policies with an existing limitador/authorino instance. Even though, currently, this wiring up works for the AuthPolicy, it is done under the assumption of a single Authorino watching all the cluster for the AuthConfig objects, which is a sub-optimal design.
+
+### Potential Scenarios
+
+* **Kuadrant supports only one instance deployed in a hard-coded namespace**
+
+No need to wire, as authorino and limitador location is well known. All policies will be linked to the same instance of Limitador/Authorino.
+
+* **Kuadrant supports only one instance deployed in a configurable namespace**
+
+Wiring is needed. The policy controllers do not know where Limitador/Authorino live. All policies will be linked to the same instance of Limitador/Authorino.
+
+* **Kuadrant supports multiple instances in a cluster**
+
+Wiring is needed. There should be a way to know the Limitador/Authorino instance only by reading the policy.
+
+# Guide-level explanation
+[guide-level-explanation]: #guide-level-explanation
+
+The proposal is to make Kuadrant to support multiple instances in a cluster.
+
+This proposal is based on a simple design decision: One gateway runs one and only one kuadrant's external auth service (Authorino) and only one kuadrant's external rate limit service (Limitador).
+
+Multiple rate limit or auth stages involving multiple instances of Authorino and Limitador can be implemented.
+Until there is a real use case and it is strictly necessary, this scenario is discarded for implementing kuadrant's protection.
+The main reason is about complexity. It is already complex enough to reason about rate limiting and auth services having a single stage. Adding multiple rate limiting stages, or hitting multiple Limitador instances in a single stage (doable with the WASM module) makes it too complex to reason about observed behavior. Currently there is no use case to require that complex scenario.
+
+A kuadrant instance includes:
+
+* One Limitador deployment instance
+* One Authorino deployment instance
+* A list of dedicated gateways. Those gateways cannot be shared between multiple kuadrant instances.
+
+![](https://i.imgur.com/QdeCYs6.png)
```

Comment:
The diagram can be seen also from the branch: https://github.com/Kuadrant/architecture/blob/wire-policies-with-backing-service/rfcs/0000-wire-policies-with-backend-services.md